### ISSUE TITLE
`extendRoutes` not working in a very confusing way

### ISSUE THREAD

**Issue by corsen2000** [nuxt/nuxt.js#299]
I am using `extendRoutes` in `nuxt.config.js` like so:

```js
router: {
  extendRoutes(routes) {
    routes.push({
      name: 'external',
      path: '/external',
      component: 'C:\\\\Users\\\\ME\\\\PROJECT\\\\external\\\\basic\\\\external-basic.vue'
    });
  }
}
```

The build runs and it generates a `router.js` file that looks correct:

```js
{
  path: "/external",
  component: _90d601e2,
  name: "external"
}
```

But, when I load `http://localhost:3000/external` in my browser, I get a 404 not found error.

When I try to start on the home page and follow a link:

```html
<nuxt-link to="/external">External Basic</nuxt-link>
```

This results in a network request to `http://localhost:3000/api/users/external`.

I'm not sure what is going on here. Does anyone know what I am doing wrong or can provide guidance on how to debug this further?

*`\\\\` is in the path due to the windows issue mentioned in #275*

Thank you.

**Comment by alexchopin** [comment]
Hi, It's a known bug, it will be fixed in the next release.

**Comment by Atinux** [comment]
@corsen2000 if your page in your nuxt project? If so, try to give it a path with `~`:

```js
router: {
  extendRoutes(routes) {
    routes.push({
      name: 'external',
      path: '/external',
      component: '~/custom-pages/external-basic.vue'
    });
  }
}
```

**Comment by corsen2000** [comment]
@Atinux I tried something similar with `~pages/about.vue`, but ran into the same issue.

In my real world project the page will not be within my nuxt project though. I am trying to add a custom route to page that another team provided to me and I installed with `NPM`. So the page is really in `node_modules`.

**Comment by corsen2000** [comment]
I've resolved my primary issue. The problem was that the Nuxt template generated a file `_id.vue` for its example which essentially created a catch all route at `root/:id`.

I was following the `extendRoutes` example from the documentation and `push` my route at the bottom of the route list. The `:id` route was preventing my route from being hit.

I'm still having the issue regarding windows file paths, but am working around that by converting my file path to unix manually.

I don't know which known issue @alexchopin was referring to. If it is not valuable to have this issue opened please close. I can file a new issue for the windows file path bug if it would be helpful.

**Comment by Atinux** [comment]
Ok I close the issue since this "bug" is resolved and for Windows we already have the #275 opened.
